### PR TITLE
Fix WSDL parser validation issues

### PR DESCRIPTION
**Pull request by vidurananayakkara**
Identitifed issues when validating WSDL XML input stream fixed

### CODE REVIEW

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java [hunk 1]
```diff
@@ -221,4 +215,87 @@ private static void registerExtensionValidators(ClassLoader classLoader) {
...
+     * Load XML data to a temporary file.
+     *
+     * @param document XML DOM
+     * @return URL of the file
+     * @throws IOException          on error writing to file
+     * @throws TransformerException on transforming error
+     */
+    private URL loadXMLToFile(Document document) throws TransformerException, IOException {
+
+
```
Comment (Shan1024) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java]: Remove unnecessary spaces.

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java [hunk 2]
```diff
@@ -221,4 +215,87 @@ private static void registerExtensionValidators(ClassLoader classLoader) {
...
+     * @param payload String XML
+     * @return XML Document
+     * @throws WSDLValidatorException on SAX, IO or parsing error
+     */
+    private Document secureParseXML(String payload) throws WSDLValidatorException {
+
+        Document document;
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        try {
+
```
Comment (Shan1024) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java]: Remove unnecessary spaces.

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator.ui/src/main/resources/web/wsdl_validator/index.jsp [hunk 3]
```diff
@@ -73,7 +73,7 @@
             document.getElementById("fileResult").innerHTML = value;
         }
 
-        if (value.search(/INVALID/g) == -1) {
+        if (value == "WSDL DOCUMENT IS VALID") {
```
Comment (imesh) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator.ui/src/main/resources/web/wsdl_validator/index.jsp]: Why do we use equal operator here? Might need to use value.equals():
http://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java [hunk 4]
```diff
@@ -221,4 +215,87 @@ private static void registerExtensionValidators(ClassLoader classLoader) {
...
+     * @param document XML DOM
+     * @return URL of the file
+     * @throws IOException          on error writing to file
+     * @throws TransformerException on transforming error
+     */
+    private URL loadXMLToFile(Document document) throws TransformerException, IOException {
+
+
+        DOMSource source = new DOMSource(document);
+        File tempFile = File.createTempFile("temp", ".txt");
```
Comment (imesh) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java]: Is this method thread safe? Is it possible for multiple threads to invoke this method at the same time? If so the temp file will get overwritten by other threads.

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java [hunk 5]
```diff
@@ -221,4 +215,87 @@ private static void registerExtensionValidators(ClassLoader classLoader) {
...
+            dbf.setFeature("http://xml.org/sax/features/validation", true);
+
+            // Build the grammar but do not use the default attributes and attribute types information it contains.
+            dbf.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+
+            // Ignore the external DTD completely.
+            dbf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+
+            DocumentBuilder db = dbf.newDocumentBuilder();
+
```
Comment (imesh) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java]: Might be better to keep less blank lines.

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/exception/WSDLValidatorException.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+public class WSDLValidatorException extends Exception {
+
+    public WSDLValidatorException(String message) {
+        super(message);
+    }
+
+    public WSDLValidatorException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
```
Comment (imesh) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/exception/WSDLValidatorException.java]: Please remove blank lines.

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/test/java/org/wso2/carbon/tools/wsdlvalidator/test/constants/XXEResourceConstants.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class for maintaining the constants for XXE XML resources.
+ */
+public class XXEResourceConstants {
+
+    private XXEResourceConstants() {
+
```
Comment (imesh) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/test/java/org/wso2/carbon/tools/wsdlvalidator/test/constants/XXEResourceConstants.java]: Please remove blank lines.

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/test/java/org/wso2/carbon/tools/wsdlvalidator/test/constants/XXEResourceConstants.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                    add(new XXEResource("outboundxxe.xml", ExceptionString.SAX_ERROR_PROCESSING_XML));
+                    add(new XXEResource("validwsdl.xml", ExceptionString.VALID_WSDL_EXCEPTION));
+                    add(new XXEResource("wafbypass.xml", ExceptionString.SAX_ERROR_PROCESSING_XML));
+                }
+            });
+
+    /**
+     * Exception constants for resource files.
+     */
+    private static class ExceptionString {
```
Comment (imesh) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/test/java/org/wso2/carbon/tools/wsdlvalidator/test/constants/XXEResourceConstants.java]: What would be the reason for using a private static class?

File: components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java (hunk 4 above)
Comment (vidurananayakkara) [inline components/wsdl-tools/wsdlvalidator/org.wso2.carbon.tools.wsdlvalidator/src/main/java/org/wso2/carbon/tools/wsdlvalidator/WsdlValidator.java]: When generating the temp file name, it is using SecureRandom which is thread safe. The file is always getting a different name.
Furthermore, createTempFile implementation is looping, generating a new file name, until the file has been created.

**Pull request merged**